lambdas/ec2/schema_enforcer.py

```diff
@@ -6,26 +6,35 @@
 import time
 import dateutil
 import re
+import os
+from warnings import warn
 
 
-def check_for_tag(ec2_instance, tag_name):
+# The `LIVE_MODE` environment variable controls if this script is actually
+# running and reaping in your AWS environment. To turn reaping on, supply
+# a value for the `LIVE_MODE` environment variable in your Lambda environment.
+LIVE_MODE = "LIVE_MODE" in os.environ
+
+# The `MINUTES_TO_WAIT` global variable is the number of minutes to wait for
+# a termination_date tag to appear for the EC2 instance. Please note that the
+# AWS Lambdas are limited to a 5 minute maximum for their total run time.
+MINUTES_TO_WAIT = 4
+
+def get_tag(ec2_instance, tag_name):
     """
     :param ec2_instance: a boto3 resource representing an Amazon EC2 Instance.
     :param tag_name: A string of the key name you are searching for.
 
-    This method returns False if the ec2 instance currently has no tags
+    This method returns None if the ec2 instance currently has no tags
     or if the tag is not found. If the tag is found, it returns the tag
     value.
     """
-    ec2_instance.load()
     if ec2_instance.tags is None:
-        return False
+        return None
```

Comment:
I changed it here to reflect more accurately that the tag did not exist, not that it was a False value.

In your example, `None` is not `False`, so that is why that logic evaluates like that. Here is a more accurate snippet of `None` evaluates with `if`:

````
>>> if None:
...     print('None is True!')
... else:
...     print('None is False!')
None is False!
````